backtrader/writer.py

```diff
@@ -102,22 +105,27 @@ def __init__(self):
         self.headers = list()
         self.values = list()
 
+    def start(self):
         # open file if needed
-        if isinstance(self.p.out, string_types):
-            self.out = open(self.p.out, 'w')
-            self.close_out = True
-        else:
-            self.out = self.p.out
-            self.close_out = self.p.close_out
+        if not hasattr(self, 'out') or not self.out:
+            if self.p.out is None:
+                self.out = sys.stdout
```

Comment:
Moving the initialization of the `self.out` member to the `start` method instead of `__init__` causes the regression in `WriterIOStream.__init__` method